src/mscorlib/shared/System/Nullable.cs

```diff
@@ -148,5 +149,15 @@ public static Type GetUnderlyingType(Type nullableType)
             }
             return null;
         }
+
+        /// <summary>
+        /// Deconstruct a <see cref="Nullable{T}"/> into a <see cref="Tuple{Boolean, T}"/>.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public static void Deconstruct<T>(this Nullable<T> nullable, out bool hasValue, out T value) where T: struct
+        {
+            hasValue = nullable.HasValue;
+            value = nullable ?? default;
```

Comment:
As @JonHanna pointed out in https://github.com/dotnet/corefx/issues/26023#issuecomment-353740823, this should be  `value = nullable.GetValueOrDefault` because of it is more efficient.

Reply:
Or `this.value` as an instance method